crates/ark/src/lsp/completions/sources/composite/keyword.rs

```diff
@@ -32,28 +37,76 @@ impl CompletionSource for KeywordSource {
 pub fn completions_from_keywords() -> anyhow::Result<Option<Vec<CompletionItem>>> {
     let mut completions = vec![];
 
-    // provide keyword completion results
-    // NOTE: Some R keywords have definitions provided in the R
-    // base namespace, so we don't need to provide duplicate
-    // definitions for these here.
-    let keywords = vec![
-        "NULL",
-        "NA",
-        "TRUE",
-        "FALSE",
-        "Inf",
-        "NaN",
-        "NA_integer_",
-        "NA_real_",
-        "NA_character_",
-        "NA_complex_",
-        "in",
+    add_bare_keywords(&mut completions);
+    add_keyword_snippets(&mut completions);
+
+    Ok(Some(completions))
+}
+
+const BARE_KEYWORDS: &[&str] = &[
+    "NULL",
+    "NA",
+    "TRUE",
+    "FALSE",
+    "Inf",
+    "NaN",
+    "NA_integer_",
+    "NA_real_",
+    "NA_character_",
+    "NA_complex_",
+    "in",
+    "next",
+    "break",
+    "repeat",
+    "function",
+];
+
+// Snippet data is a tuple of:
+// - keyword: The reserved word
+// - label: The label for the completion item, displayed on the left in the UI
+// - snippet: The snippet to be inserted
+// - label_detail.description: The description displayed on the right in the
+//     completion UI
+
+// The only case where `keyword != label` is `fun` for `function`.
+// But in the name of preserving original behaviour, this is my opening
+// move.
+const KEYWORD_SNIPPETS: &[(&str, &str, &str, &str)] = &[
+    // (keyword, label, snippet, label_detail.description)
+    (
+        "for",
+        "for",
+        "for (${1:variable} in ${2:vector}) {\n\t${0}\n}",
+        "Define a loop",
```

Comment:
You could probably define a const constructible type that is self documenting. Like you'd replace this with

```
KeywordSnippet {
  keyword: "for",
  label: "for",
  snippet: "for (${1:variable} in ${2:vector}) {\n\t${0}\n}",
  label_details_description: "Define a loop"
}
```

Which would be nicely self documenting and cleanly unpack in the loop like

```
for KeywordSnippet { keyword, label, snippet, label_details_description } in KEYWORD_SNIPPETS {
```